lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -1077,6 +1082,43 @@ def build_accessible_object_scope(associations, options = {})
       end
     end
 
+    def accessible_objects_for_operations_function(user_id, operations, options)
+      prohibition_names = []
+
+      operation_names = operations.map do |o|
+        name = operation_identifier(o)
+        # Store prohition name if not ignoring prohibitions
+        prohibition_names << prohibition_identifier(name) unless options[:ignore_prohibitions]
+        name
+      end
+
+      all_names = operation_names + prohibition_names
+      accessible_map = all_names.map { |o| [o, []] }.to_h
+
+      # Filtering should never apply to prohibitions. It is less efficient to run the function twice but still an
+      # improvement over many ad hoc ActiveRecord calls.
+      if options[:filters] && options[:ignore_prohibitions] != true
+        result = PolicyElement.accessible_objects_for_operations(user_id, operation_names, options)
+        accessible_map.merge!(result)
+        result = PolicyElement.accessible_objects_for_operations(user_id, prohibition_names, options.except(:filters))
+        accessible_map.merge!(result)
+      else
+        result = PolicyElement.accessible_objects_for_operations(user_id, all_names, options)
+        accessible_map.merge!(result)
+      end
```

Comment:
maybe something like:

```suggestion
      postgres_result = 
        if options[:filters] && options[:ignore_prohibitions] != true
          perms_map = PolicyElement.accessible_objects_for_operations(user_id, operation_names, options)
          prohibs_map = PolicyElement.accessible_objects_for_operations(user_id, prohibition_names, options.except(:filters))
          perms_map.merge(prohibs_map)
        else
          PolicyElement.accessible_objects_for_operations(user_id, all_names, options)
        end
      
      accessible_map.merge!(postgres_result)
```
?

Reply:
24b4a9d